account_invoice_fiscal_position_update/models/account_invoice.py

```diff
@@ -27,7 +27,8 @@ def fiscal_position_change(self):
         lines_without_product = []
         fp = self.fiscal_position_id
         inv_type = self.type
-        for line in self.invoice_line_ids:
+        for line in self.invoice_line_ids.filtered(
+                lambda l: not l.display_type):
             if line.product_id:
```

Comment:
Some people don't  like `filtered` by its performance, but the truth is that you have now 2 comparison at 2 different levels, so let's unify them:
```suggestion
        for line in self.invoice_line_ids:
            if line.product_id or not l.display_type:
```